sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala

```diff
@@ -71,7 +71,7 @@ private[hive] class HiveSessionState(ctx: HiveContext) extends SessionState(ctx)
   /**
    * Parser for HiveQl query texts.
    */
-  override lazy val sqlParser: ParserInterface = HiveSqlParser
+  override lazy val sqlParser: ParserInterface = HiveSqlParser(this)
```

Comment:
don't pass in the `SessionState` here if you're just gonna use the function registry. In general it's better to pass in narrower things so we don't have a tight coupling across classes.